lightning/src/ln/channel.rs

```diff
@@ -5393,15 +5421,26 @@ impl<'a, Signer: Sign, K: Deref> ReadableArgs<&'a K> for Channel<Signer>
 
 		let mut announcement_sigs = None;
 		let mut target_closing_feerate_sats_per_kw = None;
+		// Prior to supporting channel type negotiation, all of our channels were static_remotekey
+		// only, so we default to that if none was written.
+		let mut channel_type = Some(ChannelTypeFeatures::only_static_remote_key());
```

Comment:
should we do `known()` here instead?

Reply:
That's a good point - I wanted to make it explicit, though, as we may have to iterate types in the future.